service/matching/taskReader.go

```diff
@@ -131,7 +131,7 @@ dispatchLoop:
 					// Don't try to set read level here because it may have been advanced already.
 					break
 				}
-				err := tr.tlMgr.DispatchTask(ctx, task)
+				err := tr.tlMgr.engine.DispatchSpooledTask(ctx, task, tr.tlMgr.taskQueueID)
```

Comment:
this is kind of gross, sorry. I need to refactor some stuff here later and hopefully this will change then, but in the meantime it should at least work